internal/access/scim.go

```diff
@@ -0,0 +1,20 @@
+package access
+
+import (
+	"fmt"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/infrahq/infra/internal/server/data"
+	"github.com/infrahq/infra/internal/server/models"
+)
+
+func ListProviderUsers(c *gin.Context, p *data.SCIMParameters) ([]models.ProviderUser, error) {
+	// this can only be run by an access key issued for an identity provider
+	ctx := GetRequestContext(c)
```

Comment:
Originally that was part of this change, but after building out the rest of the SCIM functionality I realized that the SCIM role wasn't needed. This endpoint can only be called with an access key issued for a provider, and it can only modify the provider that the access key was issued for. In this way is acts similarly to our other `isSelf` endpoints.

Reply:
Ahh, because we query with `AccessKey.IssusedFor`, which will have a providerID in the case of these SCIM access keys, and there should never be an overlap between userIDs and providerIDs, got it!